Tests/SwiftDocCTests/Infrastructure/DocumentationContextTests.swift

```diff
@@ -3029,6 +3018,91 @@ let expected = """
+        moduleAvailableLanguages: Set<SourceLanguage>,
+        expectedArticleDefaultLanguage: SourceLanguage,
+        file: StaticString = #file,
+        line: UInt = #line
+    ) throws {
+        precondition(
+            moduleAvailableLanguages.allSatisfy { [.swift, .objectiveC].contains($0) },
+            "moduleAvailableLanguages can only contain Swift and Objective-C as languages."
+        )
+        
+        let (_, _, context) = try testBundleAndContext(copying: "MixedLanguageFramework") { url in
+            try """
+            # MyArticle
+            
+            The framework this article is documenting is available in the following languages: \
+            \(moduleAvailableLanguages.map(\.name).joined(separator: ",")).
+            """.write(to: url.appendingPathComponent("myarticle.md"), atomically: true, encoding: .utf8)
+            
+            func removeSymbolGraph(compiler: String) throws {
+                try FileManager.default.removeItem(
+                    at: url.appendingPathComponent("symbol-graphs").appendingPathComponent(compiler)
+                )
+            }
+            
+            if !moduleAvailableLanguages.contains(.swift) {
+                try removeSymbolGraph(compiler: "swift")
+            }
+            
+            if !moduleAvailableLanguages.contains(.objectiveC) {
+                try removeSymbolGraph(compiler: "clang")
+            }
+        }
+        
+        let articleNode = try XCTUnwrap(
+            context.documentationCache.first {
+                $0.key.path == "/documentation/MixedLanguageFramework/myarticle"
+            }?.value,
+            file: file,
+            line: line
+        )
+        
+        XCTAssertEqual(
+            articleNode.availableSourceLanguages,
+            moduleAvailableLanguages,
+            "Expected the article's source languages to have inherited from the module's available source languages.",
+            file: file,
+            line: line
+        )
+        
+        XCTAssertEqual(
+            articleNode.sourceLanguage,
+            expectedArticleDefaultLanguage,
+            file: file,
+            line: line
+        )
+    }
+    
+    func testArticleAvailableSourceLanguagesIsSwiftInSwiftModule() throws {
+        enableFeatureFlag(\.isExperimentalObjectiveCSupportEnabled)
+        
+        try assertArticleAvailableSourceLanguages(
+            moduleAvailableLanguages: [.swift],
+            expectedArticleDefaultLanguage: .swift
+        )
+    }
+    
+    func testArticleAvailableSourceLanguagesIsMixedLanguageInInMixedLanguageModule() throws {
+        enableFeatureFlag(\.isExperimentalObjectiveCSupportEnabled)
+        
+        try assertArticleAvailableSourceLanguages(
+            moduleAvailableLanguages: [.swift, .objectiveC],
+            expectedArticleDefaultLanguage: .swift
+        )
+    }
+    
+    func testArticleAvailableSourceLanguagesIsObjectiveCInInObjectiveCModule() throws {
```

Comment:
```suggestion
    func testArticleAvailableSourceLanguagesIsObjectiveCInObjectiveCModule() throws {
```
Same as above comment